### ISSUE TITLE
Support JSON serialisation of models with circular references

### ISSUE THREAD

**Issue by dselman** [hyperledger/composer#1536]
## Context
CTO models may contain circular references. These currently either fail to serialise to JSON, or generate duplicate data.

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by stephane-mery** [comment]
Looks like Jackson allows to deserialize/serialize graph of objects using one of the field to be used as object ID. The Java class needs to be annotated like this:
@JsonIdentityInfo(generator=ObjectIdGenerators.PropertyGenerator.class, property="ssn")

Jackson expects identified objects to be expanded only once in a given payload -probably the first time they appear in the payload- and then referenced through their ID. 

Ideally Composer would follow this way of doing so that consuming applications can easily deserialize the generated payload.

**Comment by stephane-mery** [comment]
This is what I would expect to be generated. We'll make some test to check it deserialize well.

[payload_expected.txt](https://github.com/hyperledger/composer/files/1162826/payload_expected.txt)

**Comment by markcastle3** [comment]
@dselman did you have some thoughts on how this could be fixed ?

**Comment by davidkel** [comment]
closing as no activity and no end user reported issues.
@dselman If you think this still needs to be addressed, could you create a new issue as an improvement proposal with suggested designs to solve this.
